spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/storage/GcpStorageAutoConfigurationTests.java

```diff
@@ -43,10 +44,18 @@
 /**
  * @author Artem Bilan
  */
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
-		properties = "spring.cloud.gcp.storage.auto-create-files=false")
 @RunWith(SpringRunner.class)
-public class GoogleStorageAutoConfigurationTests {
+@SpringBootTest(
+		classes = {
```

Comment:
@artembilan @snicoll if I don't add these classes, SQL auto configuration is kicking in.

However, when I add them, I'm getting a different error:

`org.springframework.context.ApplicationContextException: Unable to start EmbeddedWebApplicationContext due to missing EmbeddedServletContainerFactory bean.`

It seems that, since this is a web application, I should add another auto-configuration class to provide the missing `EmbeddedServletContainerFactory` bean. But which one?

Reply:
FFR the class I was looking for is `EmbeddedServletContainerAutoConfiguration`, but after adding it there is another configuration class that provides a required `HttpMessageConverter` and is missing.
Since I don't want to manually add every required class, I did the opposite and filtered out the conflicting classes below.

Reply:
It is unusual to test auto-configuration using `@SpringBootTest`. You should create the  context yourself. Check the auto-configuration tests in Spring Boot for more details (check the 1.5.x branch as we have a test utility on master that you won't be able to use).

Reply:
Yeah... But this one looks like an integration test. It uses `TestRestTemplate` to verify the `GoogleStorageResourceObject`. So, this one needs a Web environment. 
We might reconsider this in the future though, because I agree with Stephane it indeed will become more and more cumbersome with the when we merge more and more feature to the `autoconfigure` module.

Reply:
FWIW we're on SF5/SB2 since yesterday, so it should be possible to use that class you're mentioning.

Reply:
@snicoll since this PR aims exclusively at migrating the rest of the starters to the `autoconfigure` model, I created #313 to improve the tests separate from this PR.